asteroidsyncserviced/openweathermapparser.cpp

```diff
@@ -0,0 +1,77 @@
+#include "openweathermapparser.h"
+
+OpenWeatherMapParser::OpenWeatherMapParser(QObject *parent) : QObject(parent)
+{
+}
+
+QList<short> OpenWeatherMapParser::getWeatherId(QJsonArray list)
+{
+    QList<short> id;
+    QDateTime timestamp;
+
+    for(int i = 0; i < 5; i++) {
+        QJsonObject dayInfo = list.takeAt(i).toObject();
+        QJsonArray dayWeather = dayInfo.value("weather").toArray();
+        QJsonObject firstEntry = dayWeather.takeAt(0).toObject();
+        id << firstEntry.value("id").toInt();
+    }
+
+    return id;
+}
+
+QList<short> OpenWeatherMapParser::getTempMin(QJsonArray list)
+{
+    QList<short> tempMin;
+    QDateTime timestamp;
+    QJsonObject dayInfo = list.at(0).toObject();
+    timestamp.setTime_t(dayInfo.value("dt").toInt());
+    QString actDate = timestamp.toString("dd.MM.yyyy");
+    QJsonObject dayMain = dayInfo.value("main").toObject();
+    short int tmp = dayMain.value("temp_min").toDouble();
+    
+    for(int i = 1; i < list.size(); i++) {
+        dayInfo = list.at(i).toObject();
+        timestamp.setTime_t(dayInfo.value("dt").toInt());
+        dayMain = dayInfo.value("main").toObject();
+        if(actDate == timestamp.toString("dd.MM.yyyy") && dayMain.value("temp_min").toDouble() < tmp) {
```

Comment:
a string comparison is always quite an expensive operation (and also a bit risky as a simple space or something is always a risk), it's usually the wrong way to approach a problem. What if you'd compare QDates instead https://doc.qt.io/qt-5/qdatetime.html#date ?